src/world/mod.rs

```diff
@@ -85,7 +85,10 @@ impl<'a> EntityBuilder<'a> {
     pub fn with<T: Component>(self, c: T) -> Self {
         {
             let mut storage = self.world.write_storage();
-            storage.insert(self.entity, c);
+            if let EntityIsDead(_) = storage.insert(self.entity, c) {
+                // This is technically possible, however quite unlikely.
```

Comment:
I don't think it is, try yourself. It should not fail before the deletion is maintained.

Reply:
Oh hm, I forgot the EntityBuilder has a lifetime in it.